apps/docs/src/pages/components/tree-view.mdx

```diff
@@ -0,0 +1,433 @@
+    const { children, nodeId, label, ...others } = props;
+    const Icon = children ? () => <Folders /> : () => <Doc />;
+
+    return (
+      <HvTreeItem
+        ref={ref}
+        nodeId={nodeId}
+        label={
+          <div className="flex items-center">
+            <Icon />
+            <span style={{ flex: 1 }}>{label}</span>
+          </div>
+        }
+        {...others}
+      >
+        {children}
+      </HvTreeItem>
+    );
+  },
+);
+
+/** Render tree view items */
+const renderItem = ({ id, label, children }: TreeData) => (
+  <SimpleTreeItem key={id} nodeId={id} label={label}>
+    {children?.map(renderItem)}
+  </SimpleTreeItem>
+);
+
+export default function Demo() {
+  return (
+    <HvPanel style={{ width: 400 }}>
+      <HvTreeView aria-label="file system navigator">
+        {renderItem(treeDataObject)}
+      </HvTreeView>
+    </HvPanel>
+  );
+}
+```
+
+### Controlled
+
+The tree view can be controlled by passing in `expanded`/`onNodeToggle` and `selected`/`onNodeSelect` props to control expansion and selection state respectively.
+When using `multiSelect`, the values and callbacks are of type `string[]`, and `string` otherwise.
+
+```tsx live
+import { useState } from "react";
+
+export default function Demo() {
+  const [expanded, setExpanded] = useState<string[]>([]);
+  const [selected, setSelected] = useState<string[]>([]);
+
+  const allIds = "12345".split("");
+
+  return (
+    <div>
+      <HvButton
+        variant="secondaryGhost"
+        disabled={expanded.length === 0}
+        onClick={() => setExpanded([])}
+      >
+        Collapse All
+      </HvButton>
+      <HvButton
+        variant="secondaryGhost"
+        disabled={expanded.length >= allIds.length}
+        onClick={() => setExpanded(allIds)}
+      >
+        Expand All
+      </HvButton>
+      <HvButton
+        variant="secondaryGhost"
+        disabled={selected.length === 0}
+        onClick={() => setSelected([])}
+      >
+        Unselect All
+      </HvButton>
+      <HvButton
+        variant="secondaryGhost"
+        disabled={selected.length >= allIds.length}
+        onClick={() => setSelected(allIds)}
+      >
+        Select All
+      </HvButton>
+      <HvPanel style={{ width: 400, marginTop: 8 }}>
+        <HvTreeView
+          multiSelect
+          aria-label="file system navigator"
+          expanded={expanded}
+          selected={selected}
+          onNodeSelect={(evt, nodeIds) => setSelected(nodeIds)}
+          onNodeToggle={(evt, nodeIds) => setExpanded(nodeIds)}
+        >
+          <HvTreeItem nodeId="1" label="Applications">
+            <HvTreeItem nodeId="2" label="Calendar.app" />
+          </HvTreeItem>
+          <HvTreeItem nodeId="3" label="Documents">
+            <HvTreeItem nodeId="4" label="secret.txt" />
+            <HvTreeItem nodeId="5" label="index.js" />
+          </HvTreeItem>
+        </HvTreeView>
+      </HvPanel>
+    </div>
+  );
+}
+```
+
+### Loading data
```

Comment:
Note for later: I think this would be a good sample to move to `examples/` (and "Custom tree view" too)